core/src/main/resources/org/apache/spark/ui/static/stagepage.js

```diff
@@ -652,6 +652,38 @@ $(document).ready(function () {
             executorSummaryTableSelector.column(14).visible(dataToShow.showBytesSpilledData);
           });
 
+        // Prepare data for speculation metrics
+        $("#speculationSummaryTitle").hide()
+        $("#speculationSummary").hide()
+        var speculationSummaryInfo = responseBody.speculationSummary;
+        var speculationData;
+        if(speculationSummaryInfo) {
```

Comment:
The only difference from 73747ecb970595d49c478b0eb65 is this `if` block right?

Reply:
@sarutak Yes, that is the only difference.